apps-client/mobile/app/(tabs)/profile/users/[userId].tsx

```diff
@@ -0,0 +1,341 @@
+            <View className="flex flex-row items-center gap-4 px-4 pb-4">
+              <Avatar alt={userDetailsQuery.data.name} className="w-16 h-16">
+                <AvatarImage source={{ uri: userDetailsQuery.data.profileImage }} />
+                <AvatarPPLEFallback />
+              </Avatar>
+              <View className="flex-1 gap-2">
+                <Text className="text-base-text-default text-2xl font-heading-semibold">
+                  {userDetailsQuery.data.name}
+                </Text>
+                <View className="flex flex-row items-center gap-3">
+                  <Badge>
+                    <Text className="text-xs font-heading-semibold">
+                      {userDetailsQuery.data.roles[0] || 'สมาชิก'}
+                    </Text>
+                  </Badge>
+                  <View className="flex flex-row gap-0.5 items-center">
+                    <Icon
+                      icon={UserIcon}
+                      size={20}
+                      className="text-base-primary-default"
+                      strokeWidth={1.5}
+                    />
+                    <Text className="text-sm font-heading-semibold text-base-text-medium">
+                      {renderCount(userDetailsQuery.data.numberOfFollowers)} คน
+                    </Text>
+                  </View>
+                </View>
+              </View>
+            </View>
+            <Button
+              variant={isFollowing ? 'outline-primary' : 'primary'}
+              size="sm"
+              onPress={toggleFollow}
+              className="mx-4 mb-4"
+            >
+              <Text>{isFollowing ? 'กำลังติดตาม' : 'ติดตาม'} </Text>
+            </Button>
+          </View>
+        )}
+      </View>
+      {userDetailsQuery.data?.roles[0] === 'official' ? (
+        <Animated.FlatList
+          onScroll={scrollHandler}
+          refreshControl={<FeedRefreshControl headerHeight={16} onRefresh={onRefresh} />}
+          data={data}
+          contentContainerClassName="bg-base-bg-default"
+          renderItem={renderFeedItem}
+          ListHeaderComponent={<AnnouncementSection />}
+          ListFooterComponent={
+            <FeedFooter queryResult={userFeedInfiniteQuery} className="mt-4 mx-3" />
+          }
+          onEndReached={onEndReached}
+          onEndReachedThreshold={1}
+          showsVerticalScrollIndicator={false}
+        />
+      ) : (
+        <Animated.FlatList
+          onScroll={scrollHandler}
+          refreshControl={<FeedRefreshControl headerHeight={16} onRefresh={onRefresh} />}
+          data={data}
+          contentContainerClassName="bg-base-bg-default"
+          renderItem={renderFeedItem}
+          ListFooterComponent={
+            <FeedFooter queryResult={userFeedInfiniteQuery} className="mt-4 mx-3" />
+          }
+          onEndReached={onEndReached}
+          onEndReachedThreshold={1}
+          showsVerticalScrollIndicator={false}
+        />
+      )}
```

Comment:
is this in the requirement?

also please do ListHeaderComponent={userDetailsQuery.data?.roles[0] === 'official' ? <AnnouncementSection /> : undefined} instead